src/context/QuotesContext.tsx

```diff
@@ -0,0 +1,97 @@
+import {
+  createContext,
+  useReducer,
+  useEffect,
+  ReactNode,
+  Dispatch,
+} from "react";
+import { quotes as initialQuotes } from "../quotes";
+import { Quote } from "../types/Quote";
+
+type QuotesState = {
+  quotes: Quote[];
+  currentIndex: number;
+  history: number[];
+};
+
+
+type QuotesAction =
```

Comment:
It would be better to use enum for different action types since it's always easy to make a typo